backend/corpora/common/entities/tiledb_data.py

```diff
@@ -0,0 +1,408 @@
+        a1 = tiledb.Attr(name="X_approximate_distribution", dtype="U1")
+        a2 = tiledb.Attr(name="X_normalization_assay", dtype="U1")
+        a3 = tiledb.Attr(name="cell_count", dtype=np.int32)
+        a4 = tiledb.Attr(name="cell_type", dtype="U1")
+        a5 = tiledb.Attr(name="development_stage", dtype="U1")
+        a6 = tiledb.Attr(name="disease", dtype="U1")
+        a7 = tiledb.Attr(name="ethnicity", dtype="U1")
+        a8 = tiledb.Attr(name="is_primary_data", dtype="U1")
+        a9 = tiledb.Attr(name="name", dtype="U!")
+        a10 = tiledb.Attr(name="organism", dtype="U1")
+        a11 = tiledb.Attr(name="sex", dtype="U1")
+        a12 = tiledb.Attr(name="tissue", dtype="U1")
+
+        schema = tiledb.ArraySchema(domain=dom, sparse=True, attrs=[a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12])
+        array = location + "/datasets"
+        tiledb.Array.create(array, schema)
+
+    @staticmethod
+    def destroy_db(location):
+        """FOR TESTING PURPOSES, delete our local TileDB group."""
+        shutil.rmtree(location)
+
+    def __init__(self, location):
+        self.location = location
+
+    def create_collection(
+        self,
+        name: str = "",
+        description: str = "",
+        owner: str = "",
+        contact_name: str = "",
+        contact_email: str = "",
+        curator_name: str = "",
+        links: list = None,
+    ):
+        """Creates a collection using provided data."""
+        id = Utils.new_id()
+        with tiledb.open(self.location + "/collections", "w") as A:
+            A[id] = Utils.pack_input_data({
+                "visibility": "PRIVATE",
+                "name": name,
+                "description": description,
+                "contact_name": contact_name,
+                "contact_email": contact_email,
+                "links": links, 
+                "owner": owner,
+                "datasets": [],
+                "revision_of": "",
+                "curator_name": curator_name,
+                "created_at": time.time(),
+                "updated_at": time.time(),
+                "publisher_metadata": None
+            }, "collections")
+        return id
+
+    def get_collection(self, id):
+        """Gets a collection by its id"""
+        with tiledb.open(self.location + "/collections", 'r') as A:
+            return Utils.parse_stored_data(dict(A[id]), "collections")
+
+    def get_all_collections(self):
+        """Get all public collections"""
+        with tiledb.open(self.location + "/collections", mode="r") as A:
+            qc = tiledb.QueryCondition(f"visibility == 'PUBLIC'") # TODO: query conditions don't respect overwrites?
+            q = A.query(attr_cond=qc)
+            res = q.df[:].to_dict("records")
+            for i in res:
+                res[i] = Utils.parse_stored_data(res[i], "collections")
+            return res
+
+    def get_published_collections(self, user_id, from_date, to_date):
```

Comment:
Not urgent, but this method and the one above should be merged. Maybe put a TODO